### PR TITLE
feat: [CDS-82175]: Removed Github Action plugin for CD step group

### PR DESCRIPTION
**Pull request by ivan-83**
Removed Github Action plugin for CD Step Group.

Before changes,
Screenshot 2023-10-22 at 23 29 26

After changes,

Screenshot 2023-10-22 at 23 29 52

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54477)

### CODE REVIEW

**Comment by ivan-83** [comment]
trigger uts

**Comment by sahilhindwani-harness** [comment]
*[`332-ci-manager/service/src/main/java/io/harness/ci/execution/plan/creator/CIPipelineServiceInfoProvider.java` line 272 at r1](https://reviewable.io/reviews/harness/harness-core/54477#-Nh_Tz6Z-JtfzwSMfZSI:-Nh_Tz6Z-JtfzwSMfZSJ:bwflvta) ([raw file](https://github.com/harness/harness-core/blob/e5a1aaeb77a215537282bd4116a96cb2e1ba9366/332-ci-manager/service/src/main/java/io/harness/ci/execution/plan/creator/CIPipelineServiceInfoProvider.java#L272)):*
> ```Java
> 
>     StepInfo actionStepInfo = StepInfo.newBuilder()
>                                   .setName("Github Action plugin")
> ```

We will need to remove it from static schema as well.Could you please make the change for the same?

**Comment by ivan-83** [comment]
trigger uts

**Review by ivan-83 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54477)* status: 1 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`332-ci-manager/service/src/main/java/io/harness/ci/execution/plan/creator/CIPipelineServiceInfoProvider.java` line 272 at r1](https://reviewable.io/reviews/harness/harness-core/54477#-Nh_Tz6Z-JtfzwSMfZSI:-NhbE__u6SXuQrBBk0Ne:b-47imh7) ([raw file](https://github.com/harness/harness-core/blob/e5a1aaeb77a215537282bd4116a96cb2e1ba9366/332-ci-manager/service/src/main/java/io/harness/ci/execution/plan/creator/CIPipelineServiceInfoProvider.java#L272)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

We will need to remove it from static schema as well.Could you please make the change for the same?
</blockquote></details>

hey @sahilhindwani-harness we are getting the list of plugins from API, there is no YAML schema validation on the step group level for plugins.

**Comment by ivan-83** [comment]
trigger messagecheck

**Comment by ivan-83** [comment]
trigger messagecheck

**Comment by sahilhindwani-harness** [comment]
*[`332-ci-manager/service/src/main/java/io/harness/ci/execution/plan/creator/CIPipelineServiceInfoProvider.java` line 272 at r1](https://reviewable.io/reviews/harness/harness-core/54477#-Nh_Tz6Z-JtfzwSMfZSI:-Ni-0mN_2IgawrJ_RFDM:bq69ggi) ([raw file](https://github.com/harness/harness-core/blob/e5a1aaeb77a215537282bd4116a96cb2e1ba9366/332-ci-manager/service/src/main/java/io/harness/ci/execution/plan/creator/CIPipelineServiceInfoProvider.java#L272)):*
<details><summary><i>Previously, ivan-83 (Ivan Mijailovic) wrote…</i></summary><blockquote>

hey @sahilhindwani-harness we are getting the list of plugins from API, there is no YAML schema validation on the step group level for plugins.
</blockquote></details>

Discussed offline, we will be removing this from schema as well. Approving this PR.

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r1, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54477)* status: Some of the checks are too old. 1 file(s) to review. 1 of 2 approvals obtained.

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/plan/creator/CIPipelineServiceInfoProvider.java [hunk 1]
```diff
@@ -268,12 +268,11 @@ public List<StepInfo> getStepInfo() {
...
-        StepInfo.newBuilder()
-            .setName("Github Action plugin")
-            .setType(StepSpecTypeConstants.ACTION)
-            .setStepMetaData(StepMetaData.newBuilder().addCategory(PLUGIN).addFolderPaths("Build").build())
-            .build();
+    StepInfo actionStepInfo = StepInfo.newBuilder()
+                                  .setName("Github Action plugin")
+                                  .setType(StepSpecTypeConstants.ACTION)
+                                  .setStepMetaData(StepMetaData.newBuilder().addFolderPaths("Build").build())
+                                  .build();
```
[the remark this answers is not in the document]
Comment (vardanbansal-harness) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/plan/creator/CIPipelineServiceInfoProvider.java:275]: Reviewing based on UI changed in the screenshot, looks good to me.

**Review by vardanbansal-harness — APPROVED** [review]

**Comment by ivan-83** [comment]
trigger review

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/plan/creator/CIPipelineServiceInfoProvider.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ivan-83) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/plan/creator/CIPipelineServiceInfoProvider.java:275]: @hemanthmantri yes, it's only removing from the UI.

**Review by ivan-83 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54477)* status: Some of the checks are too old. 1 file(s) to review. 2. unresolved discussions. 

___
*[`332-ci-manager/service/src/main/java/io/harness/ci/execution/plan/creator/CIPipelineServiceInfoProvider.java` line 272 at r1](https://reviewable.io/reviews/harness/harness-core/54477#-Nh_Tz6Z-JtfzwSMfZSI:-Ni1gr5k3e_rUyuviZw5:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/e5a1aaeb77a215537282bd4116a96cb2e1ba9366/332-ci-manager/service/src/main/java/io/harness/ci/execution/plan/creator/CIPipelineServiceInfoProvider.java#L272)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

Discussed offline, we will be removing this from schema as well. Approving this PR.
</blockquote></details>

Done.

**Review by raghavharness — APPROVED** [review]

**Comment by ivan-83** [comment]
trigger uts

**Comment by ivan-83** [comment]
trigger ut2

**Review by hemanthmantri — APPROVED** [review]

**Comment by ivan-83** [comment]
trigger uts

**Comment by ivan-83** [comment]
trigger ut8

**Comment by ivan-83** [comment]
trigger ut8

**Comment by ivan-83** [comment]
trigger uts

**Pull request merged**